develop-improvements

diegoveloper opened the pull request:
working on issue #216

diegoveloper committed:
- Moved AHBottomNavigation from activity to specific fragment in order to use different type of buttons on different fragments.

diegoveloper committed:
- Profile Recommendations
- Profile Badges
- Ranking

diegoveloper committed:
- Bugs fixed on Profile -> recommendations -> categories
- SwipeRefreshLayout on contact list

PedroCarrillo commented on app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java:
```diff
@@ -20,6 +20,7 @@
 package com.belatrixsf.connect.networking.retrofit.api;
 
+import com.belatrixsf.connect.entities.Category;
 import com.belatrixsf.connect.entities.Employee;
```
mmmm

PedroCarrillo commented on app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/StarAPI.java:
```diff
@@ -37,6 +38,7 @@
 /**
  * Created by PedroCarrillo on 4/26/16.
+ * modified by dvelasquez
  */
```
mmmmmmmmmmm

diegoveloper replied:
:)

PedroCarrillo commented on app/src/main/java/com/belatrixsf/connect/services/contracts/StarService.java:
```diff
@@ -20,6 +20,7 @@
 */
 package com.belatrixsf.connect.services.contracts;
 
+import com.belatrixsf.connect.entities.Category;
```
i don't see this class being used

PedroCarrillo commented on app/src/main/java/com/belatrixsf/connect/ui/account/AccountFragment.java:
```diff
@@ -231,19 +266,14 @@ public void onSaveInstanceState(Bundle outState) {
         byte [] employeeImg = savedInstanceState.getByteArray(AccountActivity.USER_IMG_PROFILE_KEY);
-        accountPresenter.loadPresenterState(subCategoryList, employee, employeeImg);
+        accountPresenter.loadPresenterState(employee, employeeImg);
     }
```
not saving the subcategories anymore?

PedroCarrillo commented on app/src/main/java/com/belatrixsf/connect/ui/account/AccountFragment.java:
```diff
@@ -434,12 +458,12 @@ public void hideProgressIndicator() {
     public void showNoDataView() {
-        noDataTextView.setVisibility(View.VISIBLE);
+        //noDataTextView.setVisibility(View.VISIBLE);
     }
```
will always be populated? 😮

PedroCarrillo commented on app/src/main/java/com/belatrixsf/connect/ui/account/AccountPresenter.java:
```diff
@@ -87,11 +86,11 @@ public void onFailure(ServiceError serviceError) {
     }
 
+    /*
     private void loadSubCategoriesStar(boolean force) {
```
oh ok, no more subcategories

PedroCarrillo commented on app/src/main/java/com/belatrixsf/connect/ui/recommendations/RecommendationsFragment.java:
```diff
@@ -0,0 +1,221 @@
+
+    private void restoreState(Bundle savedInstanceState) {
+        List<Star> list = savedInstanceState.getParcelableArrayList(_RECOMMENDATION_LIST_KEY);
+        recommendationsPresenter.loadPresenterState(list);
```
i don't see it being saved

PedroCarrillo reviewed (commented):
Changes are good just a lot of empty spaces found in new classes and unused imports

diegoveloper committed:
- Fixed unused imports.
- Fixed unsaved states of variables.
- Event module is now using the new API ( incomplete)
